### DISCUSSION TITLE
Images with different sizes are not supported - TIFFs

### DISCUSSION THREAD

**Discussion by boukenka** [SixLabors/ImageSharp#2722] [Q&A]
There was already a discussion about it : [Support multi-sized (e.g. pyramid) TIFFs](https://github.com/SixLabors/ImageSharp/discussions/1982)
I have two TIFF files. Both are Multipage with images of different sizes.
The aim is to add the images from the second file to the first.
Unfortunately, I got the following exception: : _Images with different sizes are not supported_
Below is my code.

```C#
var resultTifPath = $"{folder}result.tif";
var convertedTifPath = $"{folder}converted.tif";
var referenceTifPath = $"{folder}48921049.TIF";
var appendedTifPath = $"{folder}44053878.TIF";

// Read the reference TIF file
using var referenceTif = await Image.LoadAsync(new DecoderOptions
{
    MaxFrames = ...
}, referenceTifPath);
Console.WriteLine($"Reference TIF : Pages: {referenceTif.Frames.Count}");

// Read the appended TIF file
using var appendedTif = await Image.LoadAsync(appendedTifPath);
Console.WriteLine($"Appended TIF : Pages: {appendedTif.Frames.Count}");

// Resize and change the bits per pixel of the appended TIF file
appendedTif.Mutate(x => x.Resize(referenceTif.Width, referenceTif.Height));
await appendedTif.SaveAsTiffAsync(convertedTifPath, new TiffEncoder
{
    BitsPerPixel = (TiffBitsPerPixel?)referenceTif.PixelType.BitsPerPixel
});

// Read the converted TIF file
using var convertedTif = await Image.LoadAsync(convertedTifPath);

// Add the frames of the converted TIF to the reference TIF
foreach (var frame in convertedTif.Frames)
{
    referenceTif.Frames.AddFrame(frame);
}
await referenceTif.SaveAsTiffAsync(resultTifPath);

// Read the result TIF file to check the process
using var resultTif = await Image.LoadAsync(resultTifPath);
Console.WriteLine($"Pages: {resultTif.Frames.Count}");
Console.ReadLine();
```

How can I do ? 
The workaround is read to first frame and convert the following frames to the same Width/Height according to the discussion.
How can it be achieved, but not manually ? 
Any help would be appreciated. 😄

**Comment by boukenka** [comment]
A solution to this problem could be found in the [v4 roadmap](https://github.com/SixLabors/ImageSharp/discussions/2533) features list. Is a preview available?

**Reply by JimBobSquarePants** [reply]
Not yet. Happy to accept PRs though.

**Comment by boukenka** [comment]


**Reply by JimBobSquarePants** [reply]
Please don't offer answers recommending other libraries. That's not constructive.

**Reply by boukenka** [reply]
Comment deleted. I don't see how this comment is not constructive. I'm not writing not to use your library. I'm just writing that for this specific problem, there is another solution. For now. It can help people until v4 is available.
